metasv/generate_sv_intervals.py

```diff
@@ -25,6 +26,22 @@ def concatenate_files(files, output):
                 outfile.write(infile.read())
 
 
+def find_softclip(aln):
+    if aln.cigar is None:
+        return None
+    soft_clips = [(i, length) for i,(op, length) in enumerate(aln.cigar) if op == 4]
+    if len(soft_clips) != 1:
+        return None
+    
+    i, soft_clip = soft_clips[0]
+    dist_L_end = sum(map(lambda x:x[1], aln.cigar[0:i]))
```

Comment:
For example, suppose the CIGAR string is 10M20D100M20I100M50S, if you are figuring out the sequence length in the read before the soft-clip, then it would be 10+100+20+100 (the initial 20D is not in the read since it is a deletion).